presto-spi/src/main/java/com/facebook/presto/spi/block/VariableWidthBlockEncoding.java

```diff
@@ -82,19 +83,15 @@ public Block readBlock(SliceInput sliceInput)
         int positionCount = sliceInput.readInt();
 
         // offsets
-        int[] offsets = new int[positionCount + 1];
-        int offset = 0;
-        for (int position = 0; position < positionCount; position++) {
-            offset += sliceInput.readInt();
-            offsets[position + 1] = offset;
-        }
+        Slice offsets = Slices.allocate((positionCount + 1) * SIZE_OF_INT);
```

Comment:
I think they're the same, but since I changed it to call `readBytes` it seemed natural to change the variable to a `Slice`